packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -693,7 +706,9 @@ abstract class DatabaseInstanceNew extends DatabaseInstanceBase implements IData
       backupRetentionPeriod: props.backupRetention?.toDays(),
       copyTagsToSnapshot: props.copyTagsToSnapshot ?? true,
       dbInstanceClass: Lazy.string({ produce: () => `db.${this.instanceType}` }),
-      dbInstanceIdentifier: props.instanceIdentifier,
+      dbInstanceIdentifier: props.instanceIdentifier === PhysicalName.GENERATE_IF_NEEDED
+        ? this.physicalName
+        : props.instanceIdentifier,
```

Comment:
Isn't all this logic already handled within `Resource` [here](https://github.com/aws/aws-cdk/blob/ecc9bf386ca088ca82a332c649f13613b9793628/packages/%40aws-cdk/core/lib/resource.ts#L157-L165)?

Shouldn't this just be 

```suggestion
      dbInstanceIdentifier: this.physicalName,
```

Reply:
Also `props.instanceIdentifier` can be upper case which is disallowed in RDS?

Reply:
I agree this was written in a confusing way. I've re-factored the code a little bit, and added a comment, hopefully it's more clear now.